flink/v1.14/flink/src/test/java/org/apache/iceberg/flink/source/reader/TestIcebergSourceReader.java

```diff
@@ -0,0 +1,191 @@
+    CombinedScanTask task =
+        ReaderUtil.createCombinedScanTask(
+            recordBatchList, TEMPORARY_FOLDER, FileFormat.PARQUET, appenderFactory);
+    IcebergSourceSplit split = IcebergSourceSplit.fromCombinedScanTask(task);
+    reader.addSplits(Arrays.asList(split));
+
+    while (readerOutput.getEmittedRecords().size() < expectedCount) {
+      reader.pollNext(readerOutput);
+    }
+
+    Assert.assertEquals(expectedCount, readerOutput.getEmittedRecords().size());
+    TestHelpers.assertRowData(
+        TestFixtures.SCHEMA,
+        recordBatchList.get(0).get(0),
+        readerOutput.getEmittedRecords().get(expectedCount - 1));
+    Assert.assertEquals(expectedCount, metricGroup.counters.get("assignedSplits").getCount());
+
+    // One more poll will get null record batch.
+    // That will finish the split and cause split fetcher to be closed due to idleness.
+    // Then next split will create a new split reader.
+    reader.pollNext(readerOutput);
+    Assert.assertEquals(expectedCount, metricGroup.counters.get("finishedSplits").getCount());
+  }
+
+  private IcebergSourceReader createReader(
+      MetricGroup metricGroup, SourceReaderContext readerContext) {
+    IcebergSourceReaderMetrics readerMetrics =
+        new IcebergSourceReaderMetrics(metricGroup, "db.tbl");
+    RowDataReaderFunction readerFunction =
+        new RowDataReaderFunction(
+            new Configuration(),
+            TestFixtures.SCHEMA,
+            TestFixtures.SCHEMA,
+            null,
+            true,
+            new HadoopFileIO(new org.apache.hadoop.conf.Configuration()),
+            new PlaintextEncryptionManager());
+    return new IcebergSourceReader<>(readerMetrics, readerFunction, readerContext);
+  }
+
+  private static class TestingMetricGroup extends UnregisteredMetricsGroup
+      implements SourceReaderMetricGroup {
+    private final Map<String, Counter> counters;
+
+    private TestingMetricGroup() {
+      this.counters = Maps.newHashMap();
+    }
+
+    /** Pass along the reference to share the map for child metric groups. */
+    private TestingMetricGroup(Map<String, Counter> counters) {
+      this.counters = counters;
+    }
+
+    @Override
+    public Counter counter(String name) {
+      Counter counter = new SimpleCounter();
+      counters.put(name, counter);
+      return counter;
+    }
+
+    @Override
+    public MetricGroup addGroup(String name) {
+      return new TestingMetricGroup(counters);
+    }
+
+    @Override
+    public MetricGroup addGroup(String key, String value) {
+      return new TestingMetricGroup(counters);
+    }
+
+    @Override
+    public OperatorIOMetricGroup getIOMetricGroup() {
+      return new TestingOperatorIOMetricGroup();
+    }
+
+    @Override
+    public Counter getNumRecordsInErrorsCounter() {
+      return new SimpleCounter();
+    }
+
+    @Override
+    public void setPendingBytesGauge(Gauge<Long> pendingBytesGauge) {}
```

Comment:
Not: I have not seen empty methods formatted this way in the Iceberg code yet. Could you please check how should they be formatted?